libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenuinputs/index.tsx

```diff
@@ -118,15 +220,53 @@ export const FloatingActionMenuInputs = (props: FloatingActionMenuInputsProps):
     return <TextField className="msla-dynamic-added-param-value-TextField" value={description} onChange={onDescriptionChange} />;
   };
 
+  const stringListValues = (schemaKey: string): string[] => {
+    const model = dynamicParameterModels.find((model) => model.schemaKey === schemaKey);
+    if (model && (model.properties as DynamicallyAddedParameterInputsTextProperties).enum) {
+      return (model.properties as DynamicallyAddedParameterInputsTextProperties).enum;
+    }
+    if (model && (model.properties as DynamicallyAddedParameterInputsArrayProperties).items) {
+      return (model.properties as DynamicallyAddedParameterInputsArrayProperties).items.enum;
+    }
+    return [];
+  };
+
+  const getStringListUpdateHandler = (schemaKey: string, newValue: string[]) => {
+    if (isDynamicParameterDropdown(schemaKey)) {
+      return onStringDropdownListUpdate(schemaKey, newValue);
+    }
+    if (isDynamicParameterMultiSelect(schemaKey)) {
+      return onStringMultiSelectListUpdate(schemaKey, newValue);
+    }
+    return undefined;
+  };
+
   const dynamicParameterModels: DynamicallyAddedParameterInputsModel[] = deserialize(
     props.initialValue,
     props.isRequestApiConnectionTrigger
   ).map((model) => ({
     ...model,
     onTitleChange: onDynamicallyAddedParameterTitleChange,
     onRequiredToggle: onDynamicallyAddedParameterRequiredToggle,
+    onStringDropdownListToggle: model.properties.type === 'string' ? onStringDropdownListToggle : undefined,
+    onStringMultiSelectListToggle: ['array', 'string'].includes(model.properties.type) ? onStringMultiSelectListToggle : undefined,
     onDelete: onDynamicallyAddedParameterDelete,
     onRenderValueField,
+    isDynamicParameterMultiSelect,
+    isDynamicParameterDropdown,
+    onStringListUpdate: getStringListUpdateHandler,
+    shouldDisplayAddDropdownOption:
+      (model.properties['x-ms-content-hint'] === 'TEXT' && !(model.properties as DynamicallyAddedParameterInputsTextProperties).enum) ??
+      undefined,
+    shouldDisplayAddMultiSelectOption:
+      (model.properties['x-ms-content-hint'] === 'TEXT' && !(model.properties as DynamicallyAddedParameterInputsArrayProperties).items) ??
```

Comment:
Why does `shouldDisplayAddMultiSelectOption` check for text type, but `shouldDisplayRemoveMultiSelectOption` checks for array?

Reply:
After changing to MultiSelect the type is modified to array as in addition to the changes to the model. This is not the case for the dropdown. So it serves as a shortcut to check since it's unique to multiselect and not dropdown or any other dynamic parameter.